addons/website_forum/models/forum.py

```diff
@@ -499,7 +499,7 @@ def _get_post_karma_rights(self):
             post.can_downvote = is_admin or user.karma >= post.forum_id.karma_downvote or post.user_vote == 1
             post.can_comment = is_admin or user.karma >= post.karma_comment
             post.can_comment_convert = is_admin or user.karma >= post.karma_comment_convert
-            post.can_view = is_admin or user.karma >= post.karma_close or (post_sudo.create_uid.karma > 0 and (post_sudo.active or post_sudo.create_uid == user))
+            post.can_view = is_admin or user.karma >= post.karma_close or (post_sudo.create_uid.karma >= 0 and (post_sudo.active or post_sudo.create_uid == user))
```

Comment:
Thanks, @lm4649, looks good!

If I read this correctly, the problem does not appear only just after writing, but any attempt to reach the post.
If so, 'd suggest to rename the commit title to `allow new user to see own pending post`.

Side note= Isn't weird to hide ALL posts to the public if karma falls below 0, instead of just the closed posts (that caused the karma slash)?
I'm thinking of removing the create_uid.karma rule... in master.

Reply:
Thanks for the review. Your reading of the problem is correct. I will change the commit title. Side note: no opinion.

Reply:
is the condition useful at all ? we already check `user.karma >= post.karma_close`
(sorry if I missed something)

or maybe we should change to
```
user.karma >= min(post.karma_close, post.karma_comment)
```

Reply:
Thank you for pointing this out. My understanding is that this condition should stay within post.can_view and taking `karma_comment` into account makes sense. I will perform the change you suggest.